plugins/pulp_rpm/plugins/migrations/0037_rpm_primary_repodata_new_location.py

```diff
@@ -0,0 +1,171 @@
+from cStringIO import StringIO
+import sys
+import os
+
+from pulp.server.db import connection
+
+if sys.version < (2, 7):
+    # We need the non-C implementation so we can shove a namespace into its internal data structure,
+    # since in python 2.6, the register_namespace function wasn't present.
+    import xml.etree.ElementTree as ET
+else:
+    import xml.etree.cElementTree as ET
+
+
+RPM_NAMESPACE = 'http://linux.duke.edu/metadata/rpm'
+PULP_PACKAGES_DIR = "Packages"
+
+# this is required because some of the pre-migration XML tags use the "rpm"
+# namespace, which causes a parse error if that namespace isn't declared.
+FAKE_XML = '<?xml version="1.0" encoding="%(encoding)s"?><faketag ' \
+           'xmlns:rpm="%(namespace)s">%(xml)s</faketag>'
+
+
+def migrate(*args, **kwargs):
+    """
+    Migrate RPMs and SRPMs location to new format "Packages/[a-z]/.*".
+
+    Remove in DRPMs "drpms/" prefix in relative_path.
+
+    :param args:   unused
+    :type  args:   list
+    :param kwargs: unused
+    :type  kwargs: dict
+    """
+
+    db = connection.get_database()
+    rpm_collection = db['units_rpm']
+    srpm_collection = db['units_srpm']
+    drpm_collection = db['units_drpm']
+
+    for rpm in rpm_collection.find({}, ['repodata', 'filename']):
+        migrate_rpm_base(rpm_collection, rpm)
+    for srpm in srpm_collection.find({}, ['repodata', 'filename']):
+        migrate_rpm_base(srpm_collection, srpm)
+    for drpm in drpm_collection.find({}, ['filename']):
+        migrate_drpm(drpm_collection, drpm)
+
+
+def migrate_drpm(collection, unit):
+    """
+    Migrate DRPMs filename field to new format.
```

Comment:
I agree that it is good to follow Fedora example in general.
We are going to have a bit different layout anyway, right?

@mhrivnak, I expect  this structure if we do not change anything for drpms, is it what you are suggesting?
```
├── drpms
│   ├── foo-1.1_1.2.drpm
│   ├── foo-1.2_1.3.drpm
│   ├── bar-3.1_4.2.drpm
│  
├── Packages
│   ├── 3
│   ├── a
│   ├── b

  [...]

│   ├── x
│   ├── y
│   └── z
└── repodata
```